village/src/App.js

```diff
@@ -1,27 +1,113 @@
-import React, { Component } from 'react';
+import React, { Component } from "react"
+import { Route, NavLink } from "react-router-dom"
+import axios from "axios"
 
-import './App.css';
-import SmurfForm from './components/SmurfForm';
-import Smurfs from './components/Smurfs';
+import "./App.css"
+import SmurfForm from "./components/SmurfForm"
+import Smurfs from "./components/Smurfs"
+import SingleSmurf from "./components/SingleSmurf"
 
 class App extends Component {
   constructor(props) {
-    super(props);
+    super(props)
     this.state = {
-      smurfs: [],
-    };
+      smurfs: []
+    }
   }
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+  componentDidMount() {
+    axios
+      .get("http://localhost:3333/smurfs")
```

Comment:
Best aproach would be to move the GET request to separate method like get_smurfs() and then call it from inside componentDidMount for modularity.